### PR TITLE
Invalid values relations update

### PR DESCRIPTION
**Pull request by fogelito**
Update relationships fixes and logic improvement & added more descriptive for error messages.

### CODE REVIEW

File: phpunit.xml [hunk 1]
```diff
@@ -7,7 +7,7 @@
     convertNoticesToExceptions="true"
     convertWarningsToExceptions="true"
     processIsolation="false"
-    stopOnFailure="false">
+    stopOnFailure="true">
```
Comment (abnegate) [inline phpunit.xml:10]: Let's revert

File: src/Database/Exception/InvalidRelationshipValue.php
Comment (abnegate) [inline src/Database/Exception/InvalidRelationshipValue.php:1]: Let's name this `InvalidRelationshipValueException`

File: src/Database/Database.php [hunk 2]
```diff
@@ -5407,4 +5421,19 @@ public function getInternalAttributes(): array
...
+     * @return bool
+     */
+    protected function arrayIsList(mixed $param): bool
+    {
+        if(!is_array($param)) {
+            return false;
+        }
+
+        //todo in php 8.1 use array_is_list
+        return \array_keys($param) === \range(0, count($param) - 1);
```
Comment (abnegate) [inline src/Database/Database.php:5436]: Let's move this check inline instead of having a separate function

File: phpunit.xml (hunk 1 above)
Comment (fogelito) [reply phpunit.xml:10]: Done!

File: src/Database/Exception/InvalidRelationshipValue.php
Comment (fogelito) [reply src/Database/Exception/InvalidRelationshipValue.php:1]: Done!

File: src/Database/Database.php (hunk 2 above)
Comment (fogelito) [reply src/Database/Database.php]: Done

**Review by abnegate — APPROVED** [review]

**Pull request merged**